dcos/prometheus-dcos.yml

```diff
@@ -33,11 +33,11 @@ scrape_configs:
   - source_labels: [__name__]
     action:        replace
     target_label:  service
-    regex:         rt:[^:]+:dst:id:_:io_l5d_marathon:([^:]+):.*
+    regex:         rt:[^:]+:dst:id.*:_:io_l5d_marathon:([^:]+):.*
```

Comment:
i haven't tested this, but i think you need to match on all non-colon chars ?
`rt:[^:]+:dst:id[^:]+:_:io_l5d_marathon:([^:]+):.*`